verification/src/solidity/compiler_fetcher.rs

```diff
@@ -104,15 +108,97 @@ impl TryFrom<(json::CompilerInfo, &Url)> for CompilerInfo {
     }
 }
 
+#[derive(Default)]
+struct RefreshableCompilerVersions {
```

Comment:
let's remove this and just include arc and mutex inside `CompilerVersions`

Reply:
then CompilerVersions::fetch_from_url will return some unnecessary arc and mutex.

in `refresh_versions` func I need new actual HashMap, I can only get it from `CompilerVersions::fetch_from_url` but it returns hashmap with mutex, which I don't need

Reply:
decided to move logic of `CompilerVersions::fetch_from_url` to separate function `try_fetch_versions`